packages/components/src/styles/dialog-styles.ts

```diff
@@ -163,29 +166,25 @@ export const getDialogTransitionJssStyle = (isVisible: boolean, slideIn: '^' | '
       ? {
           opacity: 1,
           transform: 'initial',
+          // we need to ensure that the elements inside, which get auto focused by `.showModal()`, to always be in the
+          // viewport (or off the view on the start-hand side) before the dialog transition starts otherwise the transition
+          // won't work in all cases, e.g. `dir="rtl"` and `<p-flyout position="end" />`. Because auto focus would force the
+          // focused elements to be rendered in the viewport immediately and ignore the transition.
+          ...(applyAutoFocusHack && {
```

Comment:
I think we don't need the auto focus hack, in case `slideIn === '^'`.

Reply:
`applyAutoFocusHack` is removed and `slide==='<'` is added instead.